ibis/backends/base/file/__init__.py

```diff
@@ -56,7 +56,7 @@ class BaseFileBackend(BaseBackend):
 
     database_class = FileDatabase
 
-    def connect(self, path):
```

Comment:
you cannot rename any of these methods as they r public

Reply:
You're right, I'm proposing an API change.  (See the toplevel comment I just posted.)

Reply:
@jreback The top level API here isn't changing. Every backend still has the same `connect` API by way of threading args and options through `do_connect`.

Reply:
sure it is. this is leaking the `db_connect` into the base class where its not needed. you can move the `base` changes to the sql layer i think and achieve what you want.

Reply:
Right, do_connect would be required for new backends or those that want to upgrade. I don't think I see the problem, we'd release a new major version.